### PR TITLE
Add case of updating iface

### PR DESCRIPTION
**Pull request by chloerh**
- RHEL7-99419: [update-device][bandwidth] update the QoS setting
in interface by update-device with big value, negative value, or
floor without setting qos in network

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
(1/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.average.negative_val: PASS (4.17 s)
 (2/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.average.huge_val: PASS (3.82 s)
 (3/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.average.missing: PASS (3.94 s)
 (4/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.network_qos.missing: PASS (3.80 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 16.08 s

**Review by yalzhang — APPROVED** [review]

**Review by Yingshun — APPROVED** [review]
LGTM

File: libvirt/tests/src/virtual_network/iface_update.py [hunk 1]
```diff
@@ -80,10 +82,21 @@ def run(test, params, env):
...
     vmxml_backup = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    netxml_backup = network_xml.NetworkXML.new_from_net_dumpxml(network_name)
 
     try:
+        # Prepare network
+        netxml = network_xml.NetworkXML.new_from_net_dumpxml(network_name)
+        logging.debug('Network xml before update:\n%s', netxml)
+        if del_net_bandwidth and netxml.xmltreefile.find('/bandwidth'):
+            netxml.xmltreefile.remove_by_xpath('/bandwidth')
+            netxml.xmltreefile.write()
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_update.py:97]: There is a func del_element in network_xml may work as you expected, so you can have a try on it.

Comment (chloerh) [reply libvirt/tests/src/virtual_network/iface_update.py]: Updated.

Tested with '/forward' element:

Network xml before update:
```
<?xml version='1.0' encoding='UTF-8'?>
<network>
  <name>default</name>
  <uuid>1e07df9a-5063-4587-b8ff-e435101db81a</uuid>
  <forward mode="nat">
    <nat>
      <port end="65535" start="1024" />
    </nat>
  </forward>
  <bridge delay="0" name="virbr0" stp="on" />
  <mac address="52:54:00:4a:9c:c6" />
  <ip address="192.168.124.1" netmask="255.255.255.0">
    <dhcp>
      <range end="192.168.124.254" start="192.168.124.2" />
    </dhcp>
  </ip>
</network>
```
Network xml after update:
```
<?xml version='1.0' encoding='UTF-8'?>
<network>
  <name>default</name>
  <uuid>1e07df9a-5063-4587-b8ff-e435101db81a</uuid>
  <bridge delay="0" name="virbr0" stp="on" />
  <mac address="52:54:00:4a:9c:c6" />
  <ip address="192.168.124.1" netmask="255.255.255.0">
    <dhcp>
      <range end="192.168.124.254" start="192.168.124.2" />
    </dhcp>
  </ip>
</network>
```

Working well.  Thanks for the good advice :+1:

**Comment by chloerh** [comment]
New result:
JOB LOG    : /root/avocado/job-results/job-2021-04-26T12.44-fe36170/job.log
 (1/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.average.negative_val: PASS (3.86 s)
 (2/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.average.huge_val: PASS (3.87 s)
 (3/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.average.missing: PASS (3.84 s)
 (4/4) type_specific.local.virtual_network.iface_update.negative_test.update_qos.network_qos.missing: PASS (3.92 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 15.89 s

**Review by kylazhang — APPROVED** [review]

**Pull request merged**